### ISSUE TITLE
Issue with nth-child() on rows

### ISSUE THREAD

**Issue by youradds** [dompdf/dompdf#1280]
Hi,

I'm finally getting somewhere with my design. I've opted to go more for tables, as they seem to work better.

I did come across an issue with nth-child() though. For example:

```
table.inventory td:nth-child(1) { width: 35%; }
table.inventory td:nth-child(2) { width: 10%; background: red;  }
table.inventory td:nth-child(3) { text-align: right; width: 18%; background: green; }
table.inventory td:nth-child(4) { text-align: right; width: 18%; background: orange;  }
table.inventory td:nth-child(5) { text-align: right; width: 18%; background: yellow;  }
```

```
    <table class="inventory">
        <thead>
            <tr>
                <th>Item</th>
                <th>Quantity</th>
                <th>Cost / Unit</th>
                <th>VAT</th>
                <th>Grand Total</th>
            </tr>
        </thead>
        <tbody>
            <tr class="item">
                <td>Cryptex 32Gb USB Flash Drive - Ultimate Geek Gadget!  (steampunk style) </td>
                <td>1</td>
                <td>40 GBP</td>
                <td>4.24 GBP (20%)</td>
                <td>44 GBP</td>
            </tr>
            <tr class="item">
                <td>Cryptex 32Gb USB Flash Drive - Ultimate Geek Gadget!  (steampunk style) </td>
                <td>1</td>
                <td>40 GBP</td>
                <td>4.24 GBP (20%)</td>
                <td>44 GBP</td>
            </tr>
            <tr class="item">
                <td>Cryptex 32Gb USB Flash Drive - Ultimate Geek Gadget!  (steampunk style) </td>
                <td>1</td>
                <td>40 GBP</td>
                <td>4.24 GBP (20%)</td>
                <td>44 GBP</td>
            </tr>
        </tbody>
    </table>
```

Gives you:

![2016-10-07 at 13-14-23](https://cloud.githubusercontent.com/assets/2000191/19189477/1b7e01f6-8c90-11e6-9bd4-3f0d625723a9.jpg)

**Comment by bsweeney** [comment]
Definitely different than the browser, but maybe technically OK? I dunno I'll have to look at this one some more. If you add an intermediate `tr` to your styling should work as expected.

```
table.inventory tr td:nth-child(1) { width: 35%; }
table.inventory tr td:nth-child(2) { width: 10%; background: red;  }
table.inventory tr td:nth-child(3) { text-align: right; width: 18%; background: green; }
table.inventory tr td:nth-child(4) { text-align: right; width: 18%; background: orange;  }
table.inventory tr td:nth-child(5) { text-align: right; width: 18%; background: yellow;  }
```

**Comment by youradds** [comment]
Thanks. I wonder why it works with the "tr" in there. Maybe it doesn't like missing out the structure? Whereas "normal" pages are quite happy (and sometimes even encourage it)

I ended up getting around it by just adding a .col1, .col2, etc to them - and then setting the classes on them.

BTW - do you have a way for someone to submit a template as an example for others? I'd be happy for others to make use of my design (and hopefully save some people a bit of time ;))

**Comment by bsweeney** [comment]
The reason you need the "tr" in there is due to the way the rules are parsed. Without it Dompdf is looking for the nth TD child of the table. It will only match the first five TDs period.

We don't have a template library, but that might be a good idea for more complex layouts.

**Comment by youradds** [comment]
Ah ok. Well for me, I just got around it with the .col1 .col2 etc. Good to know that for the future though.

I've attached the template I created, so that you can use it as an example for people. Would be quite nice to have a repo of different designs, so people can understand how they work (or even take a completed one, and just plop their info in.

BTW - thanks for such a cool tool! The amount of time I spent trying to find a decent PDF generator is insane! (most of them require tons of server tools, emulators, and stuff like node running in the background)

[invoice-template.txt](https://github.com/dompdf/dompdf/files/518983/invoice-template.txt)

**Comment by SVilbrandt** [comment]
Change your "nth-child" to "nth-of-type". nth-child work a bit confusing. It´s better to use "tr :nth-child" if you want to select a child element on a certain position. If you want to get a special element, you should use "tr td:nth-of-type". nth-child ignore the element before ":", it looks only for the index.

**Comment by bsweeney** [comment]
Currently in dompdf nth-child and nth-of-type are [treated the same](https://github.com/dompdf/dompdf/blob/v0.7.0/src/Css/Stylesheet.php#L578) 😞. Follow #543 for updates on fixing that.

**Comment by bsweeney** [comment]
I believe this was fixed by commit 27566fe.
